### PR TITLE
FRONT-30: Browse Feed List View

### PR DESCRIPTION
**Pull request by ArnoClaude**
## Description of Issue
[Linear issue.](https://linear.app/la-famiglia-project/issue/FRONT-30/listview-of-overview-page)
## How I implemented
Installed TanStackTable with the [React adapter](https://tanstack.com/table/v8/docs/adapters/react-table) using `npm install @tanstack/react-table`
## Other
Current status:
image

### Other
- fix types and change ESLint rules and adjust them for TypeScript
- Added `prettier-plugin-tailwindcss` for sorting of classnames and conforming to ESLint rule
- uppercase enums

### CODE REVIEW

File: .idea/vcs.xml
Comment (pczern) [inline .idea/vcs.xml:1]: remove this file, add to .gitignore

File: .eslintrc.js [hunk 1]
```diff
@@ -18,7 +26,13 @@ module.exports = {
   parser: '@typescript-eslint/parser',
 
   rules: {
-    'react/jsx-filename-extension': [1, { extensions: ['.tsx', '.jsx'] }]
+    'react/jsx-filename-extension': [1, { extensions: ['.tsx', '.jsx'] }],
```
Comment (pczern) [inline .eslintrc.js:29]: this shouldn't be turned off, we have to learn to be proper engineers

Comment (pczern) [reply .eslintrc.js:29]: even I struggle with some; but code quality increases a lot lot

File: .eslintrc.js [hunk 2]
```diff
@@ -1,5 +1,13 @@
 module.exports = {
   root: true,
+  overrides: [
+    {
+      files: ['eslintrc.js'],
+      parserOptions: {
+        sourceType: 'module'
```
Comment (pczern) [inline .eslintrc.js:7]: what is this?

File: src/constants/tableHelper.tsx
Comment (pczern) [inline src/constants/tableHelper.tsx:1]: this file should not be in /constants nothing about it is constant!!!

Comment (pczern) [reply src/constants/tableHelper.tsx:1]: move it  to the actual table, or create a table component and move there in ./components

File: src/constants/tableHelper.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }),
+    columnHelper.accessor('contributorCount', {
+      header: 'Contributors'
+    }),
+    columnHelper.accessor('pullRequestCount', {
+      header: () => 'Pull Requests'
+    }),
+    columnHelper.accessor('programmingLanguage', {
+      header: () => 'Programming Language',
+      cell: (info) => <div className="text-sky-300">{info.getValue()}</div>
```
Comment (pczern) [inline src/constants/tableHelper.tsx:31]: a <span> might be better

File: src/types/programmingLanguages.ts
Comment (pczern) [inline src/types/programmingLanguages.ts:1]: This should be at ./constants, this is constant! We don't define global types in ./src only outside

File: src/types/repositoryCategories.ts
Comment (pczern) [inline src/types/repositoryCategories.ts:1]: move to ./constants

File: src/types/repositoryShort.ts
Comment (pczern) [inline src/types/repositoryShort.ts:1]: move to the file where you define it

**Review by pczern — DISMISSED** [review]
The issue with the class not being applied is, that it is in a `./constants` folder, which is not supplied in the `tailwind.config.js`.

Make sure to delete the folder `./types` for now and move type definitions for enums to `./constants` and to where the objects are being defined

**Comment by pczern** [comment]
Can we somehow do a system like 100/200/300/400.... ? and call the colors primary/secondary/tercery? or gray/blue/yellow?
My recent commit messed the colors a little bit up, please try to implement the new naming system and fix

**Review by cnk1dev — CHANGES_REQUESTED** [review]
There still seems to be an issue with the build, please debug/ fix this:
Screenshot 2023-05-14 at 01 18 05

**Review by cnk1dev — APPROVED** [review]

**Pull request merged**